snippets/paging-cursor.js

```diff
@@ -0,0 +1,48 @@
+// the perform method of our trigger
+// ensure operation.canPaginate is true!
+
+const performWihoutAsync = (z, bundle) => {
+  return Promise.resolve()
+    .then(() => {
+      if (bundle.meta.page === 0) {
+        // first page, no need to fetch a cursor
+        return Promise.resolve();
+      } else {
+        return z.cursor.get(); // Promise<string | null>
+      }
+    })
+    .then(cursor => {
+      return z.request(
+        'https://5ae7ad3547436a00143e104d.mockapi.io/api/recipes',
+        {
+          params: { cursor: cursor } // if cursor is null, it's ignored here
+        }
+      );
+    })
+    .then(response => {
+      // need to save the cursor and return a promise, but also need to pass the data along
+      return Promise.all([response.items, z.cursor.set(response.nextPage)]);
+    })
+    .then(promises => {
+      // [items[], null]
+      return promises[0];
+    });
+};
+
+// ---------------------------------------------------
+
+const performWithAsync = async (z, bundle) => {
+  const cursor = await z.cursor.get(); // string | null
+
+  const response = await z.request(
+    'https://5ae7ad3547436a00143e104d.mockapi.io/api/recipes',
+    {
+      params: { cursor: cursor } // if cursor is null, it's ignored here
```

Comment:
> if cursor is null

Same when it's `undefined`? Since that's the more likely scenario (I think?), I'd lean towards stating that directly.

```
> null == undefined
true
> null === undefined
false
```

Reply:
I dug in here - we use node's `querystring` module. non-zero falsy values get passed in without values: 

`_limit=3&_start=3&null=&undef=&zero=0&emptyStr=`

That said, the cursor response explicitly returns `null` when there's no cursor value. I can update that doc to be more clear about what happens to null values in a querystring though, since it could (though i don't think it's supposed to) matter on the external server